### PR TITLE
add OPL emulation from Chocolate Doom

### LINKED ISSUE #102: audio: OPL emulation
I'd like to offer a choice between SDL2_Mixer music rendering and OPL emulation. It does not necessarily have to be NukedOPL, I guess the one from Dosbox does just as well.

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #102 

This needs some clean-up (e.g. do we really need midifile.c?), a new config variale and integration into the menu.

### CODE REVIEW

**Comment by rfomin** [comment]
> do we really need midifile.c?

Yes, because midifile.c does the actual parsing of the MIDI files. Also, I'm planning on using it for native Windows MIDI.

**Comment by fabiangreffrath** [comment]
I see, thanks! I consider splitting out the current music code into `i_sdlmusic.c` (call it "for symmetry") and only keep selection of the actual music backend in `i_sound.c`.

**Comment by rfomin** [comment]
> I consider splitting out the current music code into `i_sdlmusic.c` (call it "for symmetry") and only keep selection of the actual music backend in `i_sound.c`.

I was planning to do the same. Hopefully this week I will finish Win MIDI. I will name it `i_winmusic.c`.

**Comment by fabiangreffrath** [comment]
In how far will this differ from using SDL_Mixer with the native Windows backend?

**Comment by rfomin** [comment]
Music volume will work. Other than that, I intentionally use the same approach as SDL_mixer. The problem is that Win API MIDI is convoluted and low level, also I don't have access to my home computer at the moment.

**Comment by fabiangreffrath** [comment]
So, this is the approach that will render midiproc useless? Right, please go ahead and kill it.

I hope our branches won't clash too much...

**Comment by rfomin** [comment]
I make PR after your branch is merged.

**Comment by fabiangreffrath** [comment]
Thanks! 👍 Though, I cannot promise it's ready this week.

**Comment by fabiangreffrath** [comment]
OPL volume still seems a lot lower than SDL, any idea why that is?

File: Source/i_oplmusic.c [hunk 1]
```diff
@@ -0,0 +1,1787 @@
...
+
+    0x3cf, 0x3d1, 0x3d2, 0x3d4, 0x3d6, 0x3d8, 0x3da, 0x3db,   // 18
+    0x3dd, 0x3df, 0x3e1, 0x3e3, 0x3e4, 0x3e6, 0x3e8, 0x3ea,
+    0x3ec, 0x3ed, 0x3ef, 0x3f1, 0x3f3, 0x3f5, 0x3f6, 0x3f8,
+    0x3fa, 0x3fc, 0x3fe, 0x36c,
+};
+
+// Mapping from MIDI volume level to OPL level value.
+
+static const unsigned int volume_mapping_table[] = {
```
Comment (rfomin) [inline Source/i_oplmusic.c:285]: There is a volume mapping table. I'm not sure myself, and I barely remember my OPL3 soundcard from 1998. I guess this is authentic.

**Comment by fabiangreffrath** [comment]
Hm, switching back and forth between SDL and OPL in Crispy seems to reveal the same difference in volume. Can you confirm?

**Comment by rfomin** [comment]
> Hm, switching back and forth between SDL and OPL in Crispy seems to reveal the same difference in volume. Can you confirm?

Yes, it's the same. Also, it was the same in my Crispy sound system branch.

**Comment by fabiangreffrath** [comment]
So, this is ready for merging?

**Comment by rfomin** [comment]
Yes, it works fine on my system.

**Comment by fabiangreffrath** [comment]
Thanks for the feedback!

**Pull request merged**